prototype/p2p/module.go

```diff
@@ -1,56 +1,192 @@
 package p2p
 
 import (
-	"google.golang.org/protobuf/types/known/anypb"
+	"errors"
```

Comment:
We already have a copy of this file at the root directory. 

I personally think that everything should be submitted to the "non-prototype" directory. It was meant to be a snapshot of last friday.

Reply:
Ok. I suppose that's where I lost you in our last conversation and caused me to make a conflict-free PR moving things into prototype.

In either case I am happy with this going to the prototype, as this is a prototype, I want the code that will go to the non-prototype to include a few refactors I have in mind.